CHANGELOG.md

```diff
@@ -3,7 +3,90 @@
+- An issue with `opa fmt` when refs contained operators was fixed (#4105) (authored by @[jaspervdj-luminal](https://github.com/jaspervdj-luminal))
+- Fix file renaming check in optimization using `opa build` (#4154) (authored by @[davidmarne-wf](https://github.com/davidmarne-wf))
+- The `allow_net` capability was added, allowing setting limits on what hosts can be reached in built-ins like `http.send` and `net.lookup_ip_addr` ([#3665](https://github.com/open-policy-agent/opa/issues/3665))
+
+### Server
+
+- A new credential provider for AWS credential files was added ([#2786](https://github.com/open-policy-agent/opa/issues/2786)) reported by @[rgueldem](https://github.com/rgueldem)
+- The new `--tls-cert-refresh-period` flag can now be provided to `opa run`. If used with a positive duration, such as "5m" (5 minutes), 
+  "24h", etc, the server will track the certificate and key files' contents. When their content changes, the certificates will be 
+  reloaded ([#2500](https://github.com/open-policy-agent/opa/issues/2500)) reported by @[patoarvizu](https://github.com/patoarvizu)
+- A new `v1/status` endpoint was added, providing the same data as the status plugin would send to a remote endpoint ([#4089](https://github.com/open-policy-agent/opa/issues/4089))
+- The HTTP router of OPA is now exposed to the plugin manager ([#2777](https://github.com/open-policy-agent/opa/issues/2777)) authored by @[bhoriuchi](https://github.com/bhoriuchi) reported by @[mneil](https://github.com/mneil)
+- Calling `print` now works in decision masking policies
+- An unintended switch between long/regular polling on 304 HTTP status was fixed (#3926) ([#3923](https://github.com/open-policy-agent/opa/issues/3923)) authored by @[floriangasc](https://github.com/floriangasc)
+- The error message about prohibited config in the discovery plugin has been improved
+- The discovery plugin no longer panics in Trigger() if downloader is nil
+- The bundle plugin now ignores service errors for file:// resources
+- The bundle plugin file loader was updated to support directories
+- A timer to HTTP request was added to the downloader
+- The requested_by field in the logging plugin is now optional
+
+### Rego
+
+- topdown: Fix confusing error message for minus operator (#4090) ([#1643](https://github.com/open-policy-agent/opa/issues/1643))
+- ast/compile: fix print rewriting for arrays in := and head vars ([#4078](https://github.com/open-policy-agent/opa/issues/4078))
+- ast/compile: fix print rewriting for arrays in unification (#4099) ([#4078](https://github.com/open-policy-agent/opa/issues/4078))
+- ast+topdown: cache Array groundness, use for plugging (#4151) ([#3679](https://github.com/open-policy-agent/opa/issues/3679))
+- ast/compile: check arity in undefined function stage (#4059) ([#4054](https://github.com/open-policy-agent/opa/issues/4054))
```

Comment:
This one is important because it can lead to policies that previously worked to fail now. 

```rego
package policy

default p = false
p {
  x := is_blue()
  input.bar[x]
}

is_blue(fruit) = y { # doesn't use fruit
  y := input.foo
}
```
returns true with 0.35.0, (`echo '{"foo": "bar", "bar": ["bar"] }' | opa-0.35.0 eval -d policy.rego -I data.policy.p`)

and fails with this on 0.36.0:
```
rego_type_error: function data.policy.is_blue has arity 1, got 0 arguments
```

Similar issues happen with built-in functions used with the wrong argument number.

The fix is to fix your policy, I'm afraid, but it's something we should call out -- just a heads-up that previously hidden errors now explode into your face.

Reply:
Thanks! I added your example and a note.